pkg/controller/syncset/syncset_controller.go

```diff
@@ -566,6 +622,32 @@ func (r *ReconcileSyncSet) reconcileDeletedSyncSetResources(applyMode hivev1.Syn
 	return newStatusList, nil
 }
 
+func (r *ReconcileSyncSet) deleteSyncSetResources(syncSetStatus hivev1.SyncSetObjectStatus, dynamicClient dynamic.Interface, ssLog log.FieldLogger) (hivev1.SyncSetObjectStatus, error) {
+	for index, resourceStatus := range syncSetStatus.Resources {
+		itemLog := ssLog.WithField("resource", fmt.Sprintf("%s/%s", resourceStatus.Namespace, resourceStatus.Name)).
+			WithField("apiversion", resourceStatus.APIVersion).
+			WithField("kind", resourceStatus.Kind)
+		gv, err := schema.ParseGroupVersion(resourceStatus.APIVersion)
+		if err != nil {
+			syncSetStatus.Resources[index].Conditions = r.setDeletionFailedSyncCondition(syncSetStatus.Resources[index].Conditions, err)
+			return syncSetStatus, err
+		}
+		gvr := gv.WithResource(resourceStatus.Resource)
+		itemLog.Debug("deleting resource")
+		err = dynamicClient.Resource(gvr).Namespace(resourceStatus.Namespace).Delete(resourceStatus.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				itemLog.WithError(err).Error("error deleting resource")
+				// TODO: set some sort of cleanup failed condition instead of DeletionFailed
+				syncSetStatus.Resources[index].Conditions = r.setDeletionFailedSyncCondition(syncSetStatus.Resources[index].Conditions, err)
```

Comment:
same as above, err should be an error like: `fmt.Errorf("failed to delete resource: %v", err)`